src/components/DataSamplesTable.tsx

```diff
@@ -132,19 +141,39 @@ const DataSamplesTable = ({
   return (
     <>
       <ConnectFirstDialog
-        isOpen={connectToRecordDialogDisclosure.isOpen}
-        onClose={connectToRecordDialogDisclosure.onClose}
+        isOpen={isConnectToRecordDialogOpen}
+        onClose={closeDialog}
         explanationTextId="connect-to-record-body"
       />
       {selectedAction && (
-        <RecordingDialog
-          actionId={selectedAction.ID}
-          isOpen={recordingDialogDisclosure.isOpen}
-          onClose={recordingDialogDisclosure.onClose}
-          actionName={selectedAction.name}
-          onRecordingComplete={handleRecordingComplete}
-          recordingOptions={recordingOptions}
-        />
+        <>
+          <ConfirmDialog
+            isOpen={isDeleteActionConfirmOpen}
+            heading={intl.formatMessage({
+              id: "delete-action-confirm-heading",
+            })}
+            body={
+              <Text>
+                <FormattedMessage
+                  id="delete-action-confirm-text"
+                  values={{
+                    action: selectedAction.name,
+                  }}
+                />
+              </Text>
+            }
+            onConfirm={() => deleteAction(selectedAction.ID)}
+            onCancel={closeDialog}
+          />
```

Comment:
Moved delete action confirmation dialog up a level so that only one dialog would be needed instead of having one per action row. This is so that only one store state is needed to control delete confirmation dialog open state.